src/txn/txn.c

```diff
@@ -1068,6 +1083,31 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
 			prev_commit_timestamp = txn_global->commit_timestamp;
 		}
 
+	/* First check if we've made something durable in the future. */
+	update_durable_ts = false;
+	prev_durable_timestamp = 0;
+	if (candidate_durable_timestamp != 0) {
+		prev_durable_timestamp = txn_global->durable_timestamp;
+		update_durable_ts =
+		    candidate_durable_timestamp > prev_durable_timestamp;
+	}
+
+	/*
+	 * If it looks like we'll need to move the global durable timestamp,
+	 * write lock and re-check.
+	 */
```

Comment:
The comment refers to taking a write lock but the code does not do that.

Reply:
955effc

Changed to "attempt atomic cas" and also fixed the corresponding comment for the commit timestamp above.